src/components/upload-study-job-code.tsx

```diff
@@ -18,130 +17,283 @@ type SignedUrlFunc = (_job: MinimalJobInfo) => Promise<PreSignedPost>
-                </Flex>
-            </Paper>
-        )
-    }
+    const removeFile = useCallback((fileToRemove: FileWithPath) => {
+        setFileProgresses((current) => current.filter((fp) => fp.file.name !== fileToRemove.name))
+    }, [])
 
     return (
         <>
-            <Dropzone
-                w="50%"
-                onDrop={onDrop}
-                onReject={(rejections) =>
-                    notifications.show({
-                        color: 'red',
-                        title: 'rejected files',
-                        message: rejections
-                            .map(
-                                (rej) =>
-                                    `${rej.file.name} ${rej.errors.map((err) => `${err.code}: ${err.message}`).join(', ')}`,
-                            )
-                            .join('\n'),
-                    })
-                }
-                {...dzProps}
-                loading={uploadState === 'uploading'}
-                multiple={false}
-            >
-                <Stack align="center" justify="center" gap="md" mih={120} style={{ pointerEvents: 'none' }}>
-                    <Dropzone.Accept>
-                        <Upload style={{ width: rem(52), height: rem(52), color: 'var(--mantine-color-blue-6)' }} />
-                    </Dropzone.Accept>
-                    <Dropzone.Reject>
-                        <X style={{ width: rem(52), height: rem(52), color: 'var(--mantine-color-red-6)' }} />
-                    </Dropzone.Reject>
-                    <Dropzone.Idle>
-                        <UploadSimple size={32} />
-                    </Dropzone.Idle>
-                    <div>
-                        <Text size="md" mb="sm" inline>
-                            Drop files here.
-                        </Text>
-                    </div>
-                </Stack>
-                <Divider my="xl" label="Or" labelPosition="center" />
-                <Group justify="center">
-                    <FileButton onChange={setFiles} accept="image/png,image/jpeg" multiple>
-                        {(props) => (
-                            <Button {...props} variant="outline" color="#616161">
-                                Upload
-                            </Button>
-                        )}
-                    </FileButton>
+            <Paper p="xl">
+                <Text>Study Code</Text>
+                <Divider my="sm" mt="sm" mb="md" />
+                <Text mb="md">
+                    This section is key to your proposal, as it defines the analysis that will generate the results
+                    you&apos;re intending to obtain from the Member&apos;s data. Upload any necessary files to support
+                    your analysis. In this iteration, we currently support .r and .rmd files.
+                </Text>
+
+                <Group>
+                    <Stack w="30%">
+                        <Dropzone
+                            onDrop={onDrop}
```

Comment:
One main question I have around this general implementation is do we want to be uploading the files before they submit the form? or would we prefer to upload the files when they submit?

I didn't know if this is a requirement that was specific or just the way you chose to implement this 🤔 I know the cancel button had some sort of logic to deal with this aka delete any of the files that were uploaded. Perhaps this could all be avoided if they're uploaded on submit instead? Just thinking out loud